### ISSUE TITLE
Replace react-vis library for drawing charts

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger-ui#1597]
This library (https://github.com/uber/react-vis) is on the verge of being unsupported (it was recently archived, but then resurrected,  apparently with a single maintainer). We do not use a lot of features from it, mostly for the charts:
```
$ grep -rl react-vis ./packages/jaeger-ui/src | grep -v DependencyGraph
./packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.tsx
./packages/jaeger-ui/src/components/Monitor/ServicesView/operationDetailsTable/opsGraph.tsx
./packages/jaeger-ui/src/components/SearchTracePage/SearchResults/ScatterPlot.jsx
./packages/jaeger-ui/src/components/SearchTracePage/SearchResults/ScatterPlot.test.js
```
I think it should not be too difficult to replace it with something more modern an actively supported / used. I don't have a specific candidate in mind, doing a bit of research and pros/cons is part of this ticket (e.g. https://wpdatatables.com/highcharts-alternative/).

NB: I intentionally excluded DependencyGraph from this, since that could be part of another issue:
  * https://github.com/jaegertracing/jaeger-ui/issues/1466

**Comment by Yushmanth-reddy** [comment]
Hey, @yurishkuro happened to come across this issue and thought chart.js would be a better alternative for react-vis. May I work on this?

**Comment by yurishkuro** [comment]
> thought chart.js would be a better alternative

that is not a "research of pros/cons". Can you defend it?

**Comment by Yushmanth-reddy** [comment]
I have gone through the website and realized the chart is simple and clean. So thought chartjs would be a better alternative. However, I am listing down the pros and cons of chartjs.
1. Pros:
 - Chartjs is a good balance between feature-rich visualizations and efficient rendering because of its small size. 
 - Which results in low loading time and page impact for the website.
 - chartjs is compatible across all modern browsers.
 - Finally provides a wide range of charts which are entirely responsive.
2. Cons:
 - Customizations might be limited compared to D3.js
 - It is unsuitable for highly customized data visualizations (In this case, D3js would be a better option).

Please let me know whether it suits the project or should look into another library.

**Comment by yurishkuro** [comment]
Functionality is not the only criteria, we also need to compare how active the project is, how sustainable it is (how many maintainers, contributors). Also, aren't we already using D3 in plexus?

**Comment by yurishkuro** [comment]
Epoch is another d3-based module. But the D3js.org website itself recommends https://observablehq.com/plot/ for charting, built by the same team as D3.

**Comment by anikdhabal** [comment]
> Epoch is another d3-based module. But the D3js.org website itself recommends https://observablehq.com/plot/ for charting, built by the same team as D3.
yes, I think choosing Plot seems to be a good decision, as it was developed by the same team and also has an active community.

**Comment by yurishkuro** [comment]
Need to verify with CNCF that the ICS license is acceptable (https://github.com/observablehq/plot/blob/main/LICENSE)

**Comment by anikdhabal** [comment]
> Yes, it is on the allowed list https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist

Great.

**Comment by anikdhabal** [comment]
@yurishkuro I've completed most of the part of this. Can I make a PR or leave it for mentorship?
![Screenshot 2023-08-15 023820](https://github.com/jaegertracing/jaeger-ui/assets/81948346/e10ea1fa-0a4a-4429-a3da-ab448b46b620)

**Comment by yurishkuro** [comment]
Candidates who demonstrate their skills via PRs have better chances of being selected.

**Comment by anikdhabal** [comment]
@yurishkuro When working with Observable Plot, I noticed a lack of functionality compared to React Vis. For instance, there aren't straightforward or default techniques for adding events and event handlers in the plot, and there's no direct way to include tooltips that display user-set content. There are several other drawbacks as well.

I've mentioned before that there is a library called Recharts which we can use. It has over 20k stars on GitHub and is widely used by many organizations. Additionally, it has an active community. The last update was made just 2 weeks ago. The best part is that it offers similar functionality to React Vis and a similar syntax. https://recharts.org/en-US

I've completed most of it. eg:-
![Screenshot 2023-08-21 024432](https://github.com/jaegertracing/jaeger-ui/assets/81948346/f391933c-7182-44d4-ba0a-e26e27cc1bcf)
What do you think?

**Comment by yurishkuro** [comment]
recharts seems reasonable.

**Comment by jayesh9747** [comment]
@yurishkuro, after updating the history version to 5.x, how can I check if it is compatible with this repository and passes all the tests?

**Comment by coderboy-yash** [comment]
Hey @yurishkuro,

I have read the entire discussion between you and anikdhabal and noticed that he proposed using Recharts.js. After reviewing it, I found that Recharts is still actively supported, with its last update just 2 days ago and 2,619,125 weekly downloads.

If you approve, I can start replacing react-vis with Recharts.

**Comment by navaneeth-dev** [comment]
I have already started, you can check the PR discussion.

**Comment by coderboy-yash** [comment]
hey @yurishkuro i have cloned the project and then made some changes locally ,I have replaced react-vis library to recharts from this file ./packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.tsx
I am attaching some screenshots of the changes tell me if you like it or not , I am not making a pr now because I have to make some more changes to it, in the mean time if you see my work it would be helpful in my further work.

previous 
![Image](https://github.com/user-attachments/assets/79ebe58f-ac0f-4397-af72-a7ab8c9b08e2)

after changing the library to recharts

![Image](https://github.com/user-attachments/assets/20977932-09e0-4a6d-9090-09684b770645)

**Comment by coderboy-yash** [comment]
@yurishkuro please reply early so I can go ahead early with my work.

**Comment by coderboy-yash** [comment]
@navaneeth-dev are you still working on this issue or I can take this issue ?

**Comment by coderboy-yash** [comment]
> [@coderboy-yash](https://github.com/coderboy-yash) there is already PR in progress for this issue

are you talking about the pr by @navaneeth-dev or is some other pr also working on this issue. 
I am leaving this issue for now but if no one closes this issue completely, please allow me to work on this issue.

**Comment by navaneeth-dev** [comment]
Yes, I am working on it, was busy with some work. Will start again soon.
